src/main/scala/com/microsoft/azure/iot/iothub2kafka/EventHubReceiver.scala

```diff
@@ -22,33 +24,42 @@ class EventHubReceiver(val connectionString: String, val receiverConsumerGroup:
   } else {
     eventHubClient.createReceiverSync(receiverConsumerGroup, partition.toString, offset.get)
   }
-
   if (this.eventHubReceiver == null) {
     throw new IllegalArgumentException("Unable to create PartitionReceiver from the input parameters.")
   }
 
   override def close(): Unit = {
     if (this.eventHubReceiver != null) {
-      this.eventHubReceiver.close()
+      this.eventHubReceiver.synchronized {
+        this.isClosing = true
+        this.eventHubReceiver.close()
```

Comment:
For the sync logic - yes, the code in ClientEntity.close synchronizes the close. Here I am trying to synchronize the close and the receive. 
As for waiting for the future to complete, I didn't think it was necessary (as a lot of the Kafka Connect SourceTask operations can be async afaik). But might be worth doing it, so adding that now.